### PR TITLE
test file for register

### PR DESCRIPTION
**Pull request by j43333**

### CODE REVIEW

File: django_scheduled_tasks/tests/test_register.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""Example test case."""
+from django_scheduled_tasks.models import ScheduledTask
+from django_scheduled_tasks.register import register_task
+from django.test import TestCase
+from unittest.mock import patch
+
+
+class RegisterTaskTest(TestCase):
+    """Example test case, testing the wrapper function."""
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:9]:
```suggestion
    """Test case for the `register_task` wrapper."""
```

File: django_scheduled_tasks/tests/test_register.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    def test_wrapper(self):
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:11]:
```suggestion
    def test_register_task(self):
```

File: django_scheduled_tasks/tests/test_register.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        """Checks that the register is empty, and then creates an object."""
+        self.assertFalse(ScheduledTask.objects.all().exists())
+
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:14]:
```suggestion
```

Comment (LeoTurnell-Ritson) [reply django_scheduled_tasks/tests/test_register.py:14]: removing some whitspace

File: django_scheduled_tasks/tests/test_register.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        def dummy():
+            pass
+
+        self.assertEqual(ScheduledTask.objects.count(), 1)
+        st = ScheduledTask.objects.first()
+        self.assertEqual(st.func, "django_scheduled_tasks.tests.test_register.dummy")
+
+    @patch('django_scheduled_tasks.register.settings', DISABLE_SCHEDULED_TASKS=True)
+    def test_register_task_with_disable_register_task(self, mock_settings):
+        """Uses settings to disable scheduled tasks."""
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:25]:
```suggestion
        """Tests that disabling in settings correctly stops a task form being registered."""
```

File: django_scheduled_tasks/tests/test_register.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    #         pass
+
+    #     @register_task(1)
+    #     def dummy():
+    #         pass
+
+        '''self.assertEqual(ScheduledTask.objects.count(), 1)
+
+        st = ScheduledTask.objects.first()
+        self.assertEqual(st.func, "django_scheduled_tasks.tests.test_register.dummy")'''
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:55]:
```suggestion
```

Comment (LeoTurnell-Ritson) [reply django_scheduled_tasks/tests/test_register.py:55]: make sure you don't leave such commented code or notes in the push

File: django_scheduled_tasks/tests/test_register.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        """Uses settings to disable scheduled tasks."""
+        @register_task(1)
+        def dummy():
+            pass
+
+        self.assertFalse(ScheduledTask.objects.all().exists())
+
+    @patch('django_scheduled_tasks.register.sys', argv=['test'])
+    def test_resgister_task_with_skipped_args(self, mock_sys_arg):
+        """Uses system to skip task."""
```
Comment (LeoTurnell-Ritson) [inline django_scheduled_tasks/tests/test_register.py:34]:
```suggestion
        """Tests that running with `test` argument stops a task form being registered."""
```

File: django_scheduled_tasks/tests/test_register.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        self.assertFalse(ScheduledTask.objects.all().exists())
+
+    @patch('django_scheduled_tasks.register.sys', argv=['test'])
+    def test_resgister_task_with_skipped_args(self, mock_sys_arg):
+        """Tests that running with `test` argument stops a task form being registered."""
+        @register_task(1)
+        def dummy():
+            pass
+
+        self.assertFalse(ScheduledTask.objects.all().exists())
```
Comment (j43333) [inline django_scheduled_tasks/tests/test_register.py:39]:
```suggestion
        self.assertFalse(ScheduledTask.objects.all().exists())

```

File: django_scheduled_tasks/tests/test_register.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (j43333) [inline django_scheduled_tasks/tests/test_register.py:40]:
```suggestion
        self.assertFalse(ScheduledTask.objects.all().exists())
```

**Pull request merged**